crates/next-custom-transforms/src/transforms/server_actions.rs

```diff
@@ -194,19 +194,21 @@ pub fn server_actions<C: Comments>(
 /// Serializes the Server References into a magic comment prefixed by
 /// `__next_internal_action_entry_do_not_use__`.
 fn generate_server_references_comment(
-    export_names_ordered_by_reference_id: &BTreeMap<&Atom, &Atom>,
+    export_names_ordered_by_reference_id: &BTreeMap<&Atom, &ModuleExportName>,
     entry_path_query: Option<(&str, &str)>,
 ) -> String {
+    // Convert ModuleExportName to string for serialization
+    let export_map: BTreeMap<&Atom, String> = export_names_ordered_by_reference_id
+        .iter()
+        .map(|(ref_id, export_name)| (*ref_id, export_name.atom().to_string()))
+        .collect();
```

Comment:
```suggestion
    let export_map: BTreeMap<_, _> = export_names_ordered_by_reference_id
        .iter()
        .map(|(ref_id, export_name)| (*ref_id, export_name.atom()))
        .collect();
```

Always avoid, `to_string()`. That allocates a new string, which isn't necessary here at all for this temporary structure.